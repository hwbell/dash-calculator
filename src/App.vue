<template>
  <div id="app">
    <div id="content">
      <div class="space-all-row fixed-top">
        <div id="time-display">
          <TimeDisplay
            :timeDate="timeDate"
            :timeHours="timeHours"
            v-bind:showTime="true"
          />
        </div>
      </div>

      <div style="margin-top: 60px" class="row">
        <div class="col-sm-12 col-md-8">
          <Weather />
          <News />
        </div>
        <div id="tools" class="col-sm-12 col-md-4 shadow-card">
          <span id="time-converter"></span>
          <p style="margin: 10px 0">
            <strong>
              Your Dashboard Tools
            </strong>
          </p>

          <div class="left-all-row">
            <ScrollButton
              v-for="(button, ind) in navButtons"
              v-bind:key="ind"
              v-bind:target="button.target"
              v-bind:text="button.text"
              customClass="btn-outline-secondary"
            />
          </div>

          <div class=""><TimeConverter /></div>

          <div class=""><Calculator msg="Calculator" /></div>
        </div>
      </div>
    </div>

    <div class="center-all-row fixed-bottom scroll-to-top">
      <ScrollButton
        target="#app"
        text="back to top"
        customClass="btn-secondary"
      />
    </div>
  </div>
</template>

<script>
import Calculator from "./components/Calculator.vue";
import TimeDisplay from "./components/TimeDisplay.vue";
import TimeConverter from "./components/TimeConverter.vue";
import News from "./components/News.vue";
import ScrollButton from "./components/ScrollButton";
import Weather from "./components/Weather";

export default {
  name: "App",
  components: {
    Calculator,
    TimeDisplay,
    TimeConverter,
    News,
    ScrollButton,
    Weather,
  },
  created: function() {
    const moment = require("moment");

    this.timeDate = moment().format("MMMM Do YYYY");
    this.timeHours = moment().format("h:mm a");

    setInterval(() => {
      this.timeDate = moment().format("MMMM Do YYYY");
      this.timeHours = moment().format("h:mm a");
    }, 60000);
  },
  data: function() {
    return {
      timeDate: "",
      timeHours: "",
      navButtons: [
        {
          text: "calculator",
          target: "#calculator",
        },
        {
          text: "time converter",
          target: "#time-converter",
        },
      ],
    };
  },
};
</script>

<style>
#app {
  position: relative;
  width: 100%;
  /* background-color: whitesmoke; */
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  /* text-align: center; */
  color: #2c3e50;
  height: 100vh;
}
#content {
  /* margin: 40px; */
  padding: 40px;
  min-height: 100vh;

  border-radius: 6px;
  /* background-color:rgba(255, 255, 255, 0.719); */
  /* box-shadow: -2px 2px 12px 0px rgba(71, 70, 71, 0.514); */
}
#time-display {
  width: 100%;
  margin: 0;
  padding: 14px;
}
.fixed-top {
  /* padding: 10px 0; */
  background-color: white;
  box-shadow: -2px 2px 12px 0px #007e4e;
}
#tools {
  margin-top: 24px;
  border-radius: 2px;
  height: 100%;
}
/* for the abs-positioned button near the bottom of news articles */
.scroll-to-top {
  padding: 20px;
}
</style>
