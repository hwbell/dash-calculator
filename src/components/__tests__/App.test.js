import { mount } from '@vue/test-utils'
import App from '../../App.vue'

const message = 'This is a base component for us to start with!';

describe('App', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(App)
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  
  test('renders the correct elements', () => {
    expect(wrapper.findAll('#app').length).toBe(1)
    expect(wrapper.findAll('#content').length).toBe(1)
    expect(wrapper.findAll('#news-component').length).toBe(1)
    expect(wrapper.findAll('#tools').length).toBe(1)
    expect(wrapper.findAll('#calculator-component').length).toBe(1)
    expect(wrapper.findAll('#time-converter-component').length).toBe(1)
    expect(wrapper.findAll('#time-display-component').length).toBe(1)


  })

  // test(`assigns the computed value on the component properly`, () => {
  //   let revMessage = message.split('').reverse().join('');
  //   expect(wrapper.vm.reversedMessage).toBe(revMessage);
  // })  

  // test('should change the message when the greet method is called', () => {
  //   wrapper.vm.greet();
  //   expect(wrapper.vm.message).toBe('Nice to meet you!');
  // })
})