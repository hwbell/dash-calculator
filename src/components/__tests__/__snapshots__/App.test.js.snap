// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  id="app"
>
  <div
    id="content"
  >
    <div
      class="space-all-row fixed-top"
    >
      <div
        id="time-display"
      >
        <div
          id="time-display-component"
          showtime="true"
        >
          <div
            class="space-all-row"
            id="current-time"
          >
            <div
              class="app-title left-all-row"
            >
              <img
                alt="Vuejs icon"
                id="vue-image"
                src="../assets/images/vuejs.png"
              />
               
              <strong>
                uejs Dashboard
              </strong>
            </div>
             
            <div>
              <span
                class="time-date"
              >
                
        May 28th 2020, 
                <span
                  class="time-hours"
                >
                  10:11 am
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="row"
      style="margin-top: 60px;"
    >
      <div
        class="col-sm-12 col-md-8"
      >
        <div
          id="news-component"
        >
          <div
            class="news-articles-holder left-all-col"
          >
            <div
              style="width: 100%;"
            >
              <div
                class="left-all-row"
                style="width: 100%;"
              >
                <p
                  style="margin: 0px 10px 0px 0px;"
                >
                  <strong>
                    Top News since
            
                    <strong
                      style="color: rgb(0, 123, 255);"
                    >
                      2020-05-21
                    </strong>
                  </strong>
                </p>
                 
                <button
                  class="btn btn-sm btn-outline-primary"
                >
                  
          last day
        
                </button>
                 
                <button
                  class="btn btn-sm btn-outline-primary"
                >
                  
          last week
        
                </button>
                 
                <button
                  class="btn btn-sm btn-outline-primary"
                >
                  
          last month
        
                </button>
              </div>
               
              <div
                class="left-all-row"
              >
                <input
                  class="search-input"
                  type="text"
                />
                 
                <button
                  class="btn btn-sm btn-primary"
                >
                  
          search
        
                </button>
                 
                <div
                  class="spinner-border text-primary"
                  role="status"
                >
                  <span
                    class="sr-only"
                  >
                    Loading...
                  </span>
                </div>
              </div>
            </div>
             
          </div>
        </div>
      </div>
       
      <div
        class="col-sm-12 col-md-4 shadow-card"
        id="tools"
      >
        <span
          id="time-converter"
        />
         
        <p
          style="margin: 10px 0px;"
        >
          <strong>
            
            Your Dashboard Tools
          
          </strong>
        </p>
         
        <a
          href="#calculator"
        >
          <button
            class="btn btn-sm btn-outline-secondary"
          >
            
            calculator
          
          </button>
        </a>
        <a
          href="#time-converter"
        >
          <button
            class="btn btn-sm btn-outline-secondary"
          >
            
            time converter
          
          </button>
        </a>
         
        <div>
          <div
            class="top-left-all-col"
            id="time-converter-component"
          >
            <h4>
              Time Converter
            </h4>
             
            <div
              class="space-all-row row"
              id="current-unix-time"
            >
              <div
                class="left-all-col"
              >
                <p
                  style="margin: 0px;"
                >
                  <strong>
                    Unix Timestamp:
                  </strong>
                </p>
                 
                <input
                  class="unix-time-input"
                  type="text"
                />
                 
                <button
                  class="btn btn-sm btn-primary"
                >
                  
        sync time & date
      
                </button>
              </div>
            </div>
             
            <hr />
             
            <div
              class="space-all-row row"
              id="current-verbal-time"
            >
              <div
                class="left-all-col"
              >
                <p
                  style="margin: 0px;"
                >
                  <strong>
                    Time & Date:
                  </strong>
                </p>
                 
                <div
                  class="row"
                  style="margin-right: 20px;"
                >
                  <div
                    class="time-input-holder col-xs-12 col-sm-4 col-md-6 col-lg-4"
                  >
                    <p
                      style="margin: 0px; font-size: 10pt;"
                    >
                      <strong>
                        Month
                      </strong>
                    </p>
                     
                    <input
                      class="regular-time-input"
                      name="Month"
                      type="text"
                    />
                  </div>
                  <div
                    class="time-input-holder col-xs-12 col-sm-4 col-md-6 col-lg-4"
                  >
                    <p
                      style="margin: 0px; font-size: 10pt;"
                    >
                      <strong>
                        Day
                      </strong>
                    </p>
                     
                    <input
                      class="regular-time-input"
                      name="Day"
                      type="text"
                    />
                  </div>
                  <div
                    class="time-input-holder col-xs-12 col-sm-4 col-md-6 col-lg-4"
                  >
                    <p
                      style="margin: 0px; font-size: 10pt;"
                    >
                      <strong>
                        Year
                      </strong>
                    </p>
                     
                    <input
                      class="regular-time-input"
                      name="Year"
                      type="text"
                    />
                  </div>
                  <div
                    class="time-input-holder col-xs-12 col-sm-4 col-md-6 col-lg-4"
                  >
                    <p
                      style="margin: 0px; font-size: 10pt;"
                    >
                      <strong>
                        Hour
                      </strong>
                    </p>
                     
                    <input
                      class="regular-time-input"
                      name="Hour"
                      type="text"
                    />
                  </div>
                  <div
                    class="time-input-holder col-xs-12 col-sm-4 col-md-6 col-lg-4"
                  >
                    <p
                      style="margin: 0px; font-size: 10pt;"
                    >
                      <strong>
                        Minute
                      </strong>
                    </p>
                     
                    <input
                      class="regular-time-input"
                      name="Minute"
                      type="text"
                    />
                  </div>
                  <div
                    class="time-input-holder col-xs-12 col-sm-4 col-md-6 col-lg-4"
                  >
                    <p
                      style="margin: 0px; font-size: 10pt;"
                    >
                      <strong>
                        Second
                      </strong>
                    </p>
                     
                    <input
                      class="regular-time-input"
                      name="Second"
                      type="text"
                    />
                  </div>
                </div>
                 
                <span
                  id="calculator"
                />
                 
                <button
                  class="btn btn-sm btn-primary"
                >
                  
        sync unix timestamp
      
                </button>
              </div>
            </div>
          </div>
        </div>
         
        <div>
          <div
            id="calculator-component"
          >
            <h4>
              Calculator
            </h4>
             
            <div
              class="expression-window"
            >
              <div
                class="expression-text"
              >
                0Â 
              </div>
               
              <!---->
            </div>
             
            <div
              class="calculator-buttons"
            >
              <div>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
         (
      
                </button>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
        ) 
      
                </button>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
         % 
      
                </button>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
        CE
      
                </button>
              </div>
              <div>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        7
      
                </button>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        8
      
                </button>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        9
      
                </button>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
         / 
      
                </button>
              </div>
              <div>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        4
      
                </button>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        5
      
                </button>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        6
      
                </button>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
         x 
      
                </button>
              </div>
              <div>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        1
      
                </button>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        2
      
                </button>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        3
      
                </button>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
         - 
      
                </button>
              </div>
              <div>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        0
      
                </button>
                <button
                  class="calculator-button btn btn-light"
                >
                  
        .
      
                </button>
                <button
                  class="calculator-button btn btn-primary"
                >
                  
        =
      
                </button>
                <button
                  class="calculator-button btn btn-secondary"
                >
                  
         + 
      
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
